utils/hct/ExtractIRForPassTest.py

```diff
@@ -124,6 +124,9 @@ def main(args):
         passes_before, passes_after = SplitAtPass(
             all_passes, args.desired_pass, args.invocation
         )
+        if passes_after is None:
```

Comment:
an exception is raised is "after" is None, and it's checked inside the SplitAtPass function. So, if you wanted to accompany this exception with an error message, wouldn't it be more proper to surround the SplitAtPass invocation with a try catch block?

Reply:
Otherwise, you might as well move this sys.exit invocation into the SplitAtPass function itself.

Reply:
I don't think I understand your first sentence here. I didn't want to add another try block. In practice, you will see the generic catch that says "something went wrong!" and then you'll see the specific error. That gave me all I wanted.

Reply:
Yeah, this is a nit. I was just noting that instead of `raise Exception(f"no such pass: {pass_name}")` and then going into your new if condition, you could replace the exception with `sys.exit("\nPass {} not found!\nUse dxc with -Odump to see the available passes.\n".format(args.desired_pass))`
Would that be unwanted?

Reply:
Dang, I already had an alternate change to throw a useful error in this case [here](https://github.com/tex3d/DirectXShaderCompiler/commit/302292df34b6ecb50b8c1c768c9c1bc5afbd5a79), but hadn't put up a PR.

Reply:
Wait, that's not true, I did have a PR here #6325.

Reply:
Looking at a prior change which introduced an exception when the pass wasn't found, it appears this change wouldn't be able to work, since `SplitAtPass` [would have raised an exception](https://github.com/microsoft/DirectXShaderCompiler/pull/6639/files/0b42af4a0bb9e00f3b7e5c65ced49ff4f9c36034#diff-eb56348534bb00cde78f36c136a36ba14c96a1af924f5ccb8dd93464c5a9b131R89-R90) before returning `None` for `passes_after`.

That change went in 2 weeks ago from this PR: #6519.